src/Orleans/Core/ClientBuilder.cs

```diff
@@ -115,6 +120,16 @@ private static void AddDefaultServices(HostBuilderContext context, IServiceColle
             services.TryAddSingleton<CodeGeneratorManager>();
             services.TryAddSingleton<IInternalClusterClient, ClusterClient>();
             services.TryAddFromExisting<IClusterClient, IInternalClusterClient>();
+            services.TryAddSingleton<ITypeResolver, CachedTypeResolver>();
+
+            // Application parts
+            var parts = context.GetApplicationPartManager();
+            services.TryAddSingleton<ApplicationPartManager>(parts);
+            parts.AddApplicationPart(typeof(RuntimeVersion).Assembly);
+            parts.AddFeatureProvider(new BuiltInTypesSerializationFeaturePopulator());
+            parts.AddFeatureProvider(new AssemblyAttributeFeatureProvider<GrainInterfaceFeature>());
+            parts.AddFeatureProvider(new AssemblyAttributeFeatureProvider<GrainClassFeature>());
```

Comment:
Does the client builder care about concrete grain implementations?

Reply:
Seems like it doesn't. I wanted everything symmetrical, Maybe using a common "AddDefaultFeatureProviders" method, but I can remove it.